### PR TITLE
Removed safari compatibility check (issue #9824)

### LINKED ISSUE #9824: Enable range requests in Safari (both OSX and IOS)
Attach (recommended) or Link to PDF file here: https://mozilla.github.io/pdf.js/web/compressed.tracemonkey-pldi-09.pdf

Configuration:
- Web browser and its version: Safari 11.1 (OSX), Safari 11.0 (IOS)
- Operating system and its version: OSX 10.13.4, IOS 11.4
- PDF.js version: latest from master
- Is a browser extension: no

Steps to reproduce the problem:
1. Start dev server with gulp server
2. Visit http://localhost:8888/web/viewer.html
2. Observe network tab for 206 requests

What is the expected behavior? (add screenshot)
The document is loaded using range requests

What went wrong? (add screenshot)
The document is loaded with a single request. It looks like range requests were explicitly disabled by this code [here](https://github.com/mozilla/pdf.js/blob/3b07147d98e376a774316288647d421b80b589c1/src/display/api_compatibility.js#L39):
```
  // Support: Safari 6.0+, iOS
  (function checkRangeRequests() {
    // Safari has issues with cached range requests, see issue #3260.
    // Last tested with version 6.0.4.
    if (isSafari || isIOS) {
      compatibilityParams.disableRange = true;
      compatibilityParams.disableStream = true;
    }
  })(); 
```
It feels a bit outdated. I've just tested pdfjs without this code in the specified browsers and it worked just fine. Is there a reason for keeping this code?

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

### PR DESCRIPTION
**Pull request by EugeneSqr**
Fixes #9824

### CODE REVIEW

File: src/display/api_compatibility.js [hunk 1]
```diff
@@ -35,16 +32,6 @@ if (typeof PDFJSDev === 'undefined' || PDFJSDev.test('GENERIC')) {
     }
   })();
 
-  // Support: Safari 6.0+, iOS
-  (function checkRangeRequests() {
-    // Safari has issues with cached range requests, see issue #3260.
-    // Last tested with version 6.0.4.
-    if (isSafari || isIOS) {
-      compatibilityParams.disableRange = true;
-      compatibilityParams.disableStream = true;
```
Comment (Snuffleupagus) [inline src/display/api_compatibility.js]: With these now being removed, all occurrences of `apiCompatibilityParams.disableRange`/`apiCompatibilityParams.disableStream` throughout the code-base will always be `undefined` and ought to be removed as well.

See also https://github.com/mozilla/pdf.js/wiki/Squashing-Commits

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Looks good. Thank you for improving this!

**Pull request merged**